sdk/eventhubs/azure-messaging-eventhubs-track1-perf/src/main/java/com/azure/messaging/eventhubs/perf/SendEventBatchPartitionTest.java

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs.perf;
+
+import com.azure.messaging.eventhubs.perf.core.EventHubsPerfStressOptions;
+import com.azure.messaging.eventhubs.perf.core.ServiceBatchTest;
+import com.microsoft.azure.eventhubs.BatchOptions;
+import com.microsoft.azure.eventhubs.EventData;
+import com.microsoft.azure.eventhubs.EventDataBatch;
+import com.microsoft.azure.eventhubs.PartitionSender;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Runs the Send Events Batch Performance Test for EventHubs.
+ */
+public class SendEventBatchPartitionTest extends ServiceBatchTest<EventHubsPerfStressOptions> {
+    private final BatchOptions batchOptions;
+    private EventDataBatch eventDataBatch;
+    private PartitionSender partitionSender;
+
+    /**
+     * Instantiates the instance of the Send Event Batch Partition test.
+     *
+     * @param options the options bag to use for performance testing.
+     * @throws Exception when an error occurs when creating event batch.
+     */
+    public SendEventBatchPartitionTest(EventHubsPerfStressOptions options) throws Exception {
+        super(options);
+        batchOptions = new BatchOptions();
+        BatchOptions batchOptions = new BatchOptions();
+
+        if (options.getBatchSize() != null) {
+            batchOptions.maxMessageSize = options.getBatchSize();
+        }
+
+        if (options.getPartitionKey() != null) {
+            batchOptions.partitionKey = options.getPartitionKey();
+        }
+
+    }
+
+    @Override
+    public int runBatch() {
+        partitionSender.send(eventDataBatch);
+        return options.getEvents();
+    }
+
+    @Override
+    public Mono<Integer> runBatchAsync() {
+        return Mono.fromFuture(partitionSender.send(eventDataBatch))
```

Comment:
If you call runBatchAsync multiple times, does it send the batch multiple times? or does it replay the result from the future?

Reply:
sends multiple times